examples/08_experimental/surface_image_and_maskers.py

```diff
@@ -0,0 +1,159 @@
+"""A short demo of the surface images & maskers
+
+copied from the nilearn sandbox discussion, to be transformed into tests &
+examples
+
+"""
+from typing import Optional, Sequence
+
+from matplotlib import pyplot as plt
+
+from nilearn import plotting
+from nilearn.experimental import surface
+
+
+def plot_surf_img(
+    img: surface.SurfaceImage,
+    parts: Optional[Sequence[str]] = None,
+    mesh: Optional[surface.PolyMesh] = None,
+    **kwargs,
+) -> plt.Figure:
+    if mesh is None:
+        mesh = img.mesh
+    if parts is None:
+        parts = list(img.data.keys())
+    fig, axes = plt.subplots(
+        1,
+        len(parts),
+        subplot_kw={"projection": "3d"},
+        figsize=(4 * len(parts), 4),
+    )
+    for ax, mesh_part in zip(axes, parts):
+        plotting.plot_surf(
+            mesh[mesh_part],
+            img.data[mesh_part],
+            axes=ax,
+            title=mesh_part,
+            **kwargs,
+        )
+    assert isinstance(fig, plt.Figure)
+    return fig
+
+
+img = surface.fetch_nki()[0]
+print(f"NKI image: {img}")
+
+masker = surface.SurfaceMasker()
+masked_data = masker.fit_transform(img)
+print(f"Masked data shape: {masked_data.shape}")
+
+mean_data = masked_data.mean(axis=0)
+mean_img = masker.inverse_transform(mean_data)
+print(f"Image mean: {mean_img}")
+
+plot_surf_img(mean_img)
+plotting.show()
+
+###############################################################################
+# ### Connectivity with a surface atlas and `SurfaceLabelsMasker`
+
+from nilearn import connectome, plotting
+
+img = surface.fetch_nki()[0]
+print(f"NKI image: {img}")
+
+labels_img, label_names = surface.fetch_destrieux()
+print(f"Destrieux image: {labels_img}")
+plot_surf_img(labels_img, cmap="gist_ncar", avg_method="median")
+
+labels_masker = surface.SurfaceLabelsMasker(labels_img, label_names).fit()
+masked_data = labels_masker.transform(img)
+print(f"Masked data shape: {masked_data.shape}")
+
+connectome = (
+    connectome.ConnectivityMeasure(kind="correlation").fit([masked_data]).mean_
+)
+plotting.plot_matrix(connectome, labels=labels_masker.label_names_)
+
+plotting.show()
+
+
+###############################################################################
+# ### Using the `Decoder`
+
+import numpy as np
+
+from nilearn import decoding, plotting
+from nilearn._utils import param_validation
+
+###############################################################################
+# The following is just disabling a couple of checks performed by the decoder
+# that would force us to use a `NiftiMasker`.
+
+
+def monkeypatch_masker_checks():
```

Comment:
Eventually what would be the suggested solution(s) here? Adjusting these functions to accept the surface masker or writing new separate functions to check the surface masker if such object is given to the decoder?

Reply:
I think `_check_embedded_nifti_masker` would have to be adapted (and maybe renamed) so that it can also work for surface maskers. for example if the passed estimator's mask_img is a surface image a surface masker would be returned.

it would have been better if the decoder objects were passed a masker rather than all the masker's parameters but I guess it is too late to change that